saw-core/src/Verifier/SAW/SharedTerm.hs

```diff
@@ -1170,20 +1211,25 @@ scLambdaList _ [] rhs = return rhs
 scLambdaList sc ((nm,tp):r) rhs =
   scLambda sc nm tp =<< scLambdaList sc r rhs
 
+-- | Create a (possibly dependent) function given a parameter name, parameter
+-- type (as a term), and a body.
```

Comment:
Again, should mention index convention: Index 0 is bound in the "body" term. We should probably also add docstrings to the individual `Term` arguments